rclpy/test/test_serialization.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from rclpy.serialization import deserialize_message
+from rclpy.serialization import serialize_message
+
+from test_msgs.message_fixtures import get_test_msg
+from test_msgs.msg import Arrays
+from test_msgs.msg import BasicTypes
+from test_msgs.msg import BoundedSequences
+from test_msgs.msg import Builtins
+from test_msgs.msg import Constants
+from test_msgs.msg import Defaults
+from test_msgs.msg import Empty
+from test_msgs.msg import MultiNested
+from test_msgs.msg import Nested
+from test_msgs.msg import Strings
+from test_msgs.msg import UnboundedSequences
+from test_msgs.msg import WStrings
+
+test_msgs = [
+  (get_test_msg('Arrays'), Arrays),
+  (get_test_msg('BasicTypes'), BasicTypes),
+  (get_test_msg('BoundedSequences'), BoundedSequences),
+  (get_test_msg('Builtins'), Builtins),
+  (get_test_msg('Constants'), Constants),
+  (get_test_msg('Defaults'), Defaults),
+  (get_test_msg('Empty'), Empty),
+  (get_test_msg('MultiNested'), MultiNested),
+  (get_test_msg('Nested'), Nested),
+  (get_test_msg('Strings'), Strings),
+  (get_test_msg('UnboundedSequences'), UnboundedSequences),
+  (get_test_msg('WStrings'), WStrings),
+]
+
+
+@pytest.mark.parametrize('msgs,msg_type', test_msgs)
+def test_serialize_deserialize(msgs, msg_type):
+    """Test message serialization/deserialization."""
+    for msg in msgs:
+        msg_serialized = serialize_message(msg)
+        msg_deserialized = deserialize_message(msg_serialized, msg_type)
+        assert msg == msg_deserialized
+
+
+# TODO(jacobperron): Fix this failing test.
+#                    I'm pretty sure that during (de)serialization we convert to a C float before
+#                    converting to a PyObject. This causes a loss of precision.
+# def test_set_float32():
+#     msg = BasicTypes()
+#     msg.float32_value = 3.14
```

Comment:
Maybe try a float value which can be represented without rounding, e.g. 1.125?

Reply:
I can confirm a value that doesn't require rounding, like 1.125, works fine. Do you think this TODO is still valid though?

Reply:
The comment that the value will be stored in a C type still applies. I am just not sure what the TODO should be about?

Reply:
Maybe there's nothing we can do about this. I could instead update the test to do the rounding in Python so it passes.

Reply:
I've replaced the TODO with a test that illustrates what could happen with float32 types: 668c9b9